lib/transaction.ex

```diff
@@ -31,6 +31,73 @@ defmodule Elixium.Transaction do
     %{outputs: outputs}
   end
 
+  @doc """
+  Creates a correct tx id
+  """
+  @spec create_tx_id(List) :: String.t()
+  def create_tx_id(tx) do
+    calculate_hash(tx)
+    |> Utilities.sha_base16()
+  end
+
+  @doc """
+  Creates a singature list
+  """
+  @spec create_sig_list(List, Map) :: List
+  def create_sig_list(inputs, transaction) do
+    digest = Elixium.Transaction.signing_digest(transaction)
+    inputs
+    |> Enum.uniq_by(& &1.addr)
+    |> Enum.map(fn %{addr: addr} ->
+      priv = Elixium.KeyPair.get_priv_from_file(addr)
+      sig = Elixium.KeyPair.sign(priv, digest)
+      {addr, sig}
+    end)
+  end
+
+
+  @doc """
+  Take the correct amount of Utxo's to send the alloted amount in a transaction.
+  """
+  @spec take_necessary_utxos(List, Decimal) :: function
+  def take_necessary_utxos(utxos, amount), do: take_necessary_utxos(utxos, [], amount)
+
+  @spec take_necessary_utxos(List, List, Decimal) :: List
+  def take_necessary_utxos(utxos, chosen, amount) do
+    if D.cmp(amount, 0) == :gt do
+      if utxos == [] do
+        :not_enough_balance
+      else
+        [utxo | remaining] = utxos
+        take_necessary_utxos(remaining, [utxo | chosen], D.sub(amount, utxo.amount))
+      end
+    else
+      chosen
+    end
+  end
+
+
+  @doc """
+  Creates a current time stamp for transaction building
+  """
+  @spec create_timestamp :: String.t()
+  def create_timestamp, do: DateTime.utc_now |> DateTime.to_string
+
+  @doc """
+  # Since a UTXO is fully used up when we put it in a new transaction, we must create a new output
+  # that credits us with the change
+  """
+  @spec create_designations(Map, Decimal, Decimal, String.t(), List) :: List
+  def create_designations(inputs, amount, desired_fee, return_address, prev_designations) do
```

Comment:
Take a look at the new transaction.ex file. There's a function there to create a transaction based on designations (and it takes into account change and fees) which I'm pretty sure makes this function obsolete

Reply:
Updated & Updated Tests & Functions